### PR TITLE
#105: fixed issue with organizations with no acronym not being pulled…

### PR DESCRIPTION
**Pull request by guillaumev**
… from HRinfo into HID

### CODE REVIEW

File: sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc [hunk 1]
```diff
@@ -86,18 +86,16 @@ function hid_organizations_autocomplete($org_names, $extended = FALSE) {
   // Search for hr_organization taxonomy terms, where the term name or
   // organization field value match the given search string.
   $query = db_select('taxonomy_term_data', 't');
-  $query->leftJoin('field_data_field_acronym', 'fdfa', '(t.tid = fdfa.entity_id AND fdfa.entity_type = :fdfa_type)', array(':fdfa_type' => 'taxonomy_term'));
+  $query->leftJoin('field_data_field_acronym', 'fdfa', '(t.tid = fdfa.entity_id AND fdfa.entity_type = :fdfa_type AND fdfa.language = \'en\')', array(':fdfa_type' => 'taxonomy_term'));
```
Comment (pcambra) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: We might want to change this for an EFQ instead?

Comment (guillaumev) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: Would be great, but I don't think I will have time to do it, at least not soon, so unless you have time to do it, maybe we could merge this and change it to an EFQ later...

**Comment by guillaumev** [comment]
@pcambra Could this be merged and deployed asap ? We are getting support requests on the HID side because this change hasn't been deployed yet... Thank you.

**Comment by pcambra** [comment]
I will rewrite it and will get on Monday's release

**Comment by pcambra** [comment]
Being this an autocomplete function, I'm not sure why we're calling it from outside and dealing with CORS.

Is this being used for autocomplete but also as an endpoint for HID?

**Comment by guillaumev** [comment]
It's being used for autocomplete only, within HID, when chosing an organization...

**Comment by pcambra** [comment]
if it's an endpoint, it should probably be managed by RESTful instead, shouldn't it?

**Comment by guillaumev** [comment]
Ideally yes, but it would mean a fairly big change in the HID app, which for now there is no time to work on (or at least it's not a priority)

**Comment by pcambra** [comment]
Do you use the case of empty search terms?, meaning having $org_names empty

**Comment by guillaumev** [comment]
No

**Comment by pcambra** [comment]
@guillaumev please review and merge if you think is OK

File: sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc [hunk 2]
```diff
@@ -67,71 +79,70 @@ function hid_profiles_settings_form($form, &$form_state) {
...
+  if (count($data) < $limit) {
+    $query = new EntityFieldQuery();
+    $results = $query
+      ->entityCondition('entity_type', 'taxonomy_term')
+      ->entityCondition('bundle', 'hr_organization')
+      ->fieldCondition('field_acronym', 'value', db_like($org_names), $operator)
+      ->propertyCondition('tid', array_keys($data), 'NOT IN')
+      ->range(0, $limit)
+      ->execute();
+    $data += reset($results);
```
Comment (guillaumev) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: @pcambra I'm getting an "unsupported operand types" on this line. Did you test it in your env ? If so and if it works, which version of PHP are you running ?

Comment (pcambra) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: using hr_docker...

Comment (guillaumev) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: But are you having the same issue ? Did you try opening a URL like http://www.hrinfo.vm/hid/organizations/extended/autocomplete/British ?

Comment (pcambra) [inline sites/all/modules/hid/hid_profiles/hid_profiles.pages.inc]: Well, I see now that the way hr_docker is built does not ensure same versions of PHP (and probably any other package). 5.5.9-1ubuntu4.11 vs 5.3.29
I will use an array_merge here instead and work on having at least the right PHP version there.

**Comment by pcambra** [comment]
@guillaumev I believe you're not in gitter anymore, could you recheck the latest change? thanks

**Pull request merged**